### PR TITLE
CA-16 resource service

### PR DESCRIPTION
**Pull request by mrkvcstms**

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/Kingdom.java [hunk 1]
```diff
@@ -63,6 +63,16 @@ public Kingdom(AppUser appUser) {
...
 
+  // region Getters
+  public Resource getGold() {
+    return this.getResources().get(0);
+  }
+
+  public Resource getFood() {
+    return this.getResources().get(1);
+  }
+  // endregion Getters
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/Kingdom.java]: should be hashmap

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceService.java [hunk 2]
```diff
@@ -11,4 +12,5 @@
 
   public List<Resource> createDefaultResources();
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceService.java]: hashmap

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImpl.java [hunk 3]
```diff
@@ -43,5 +49,38 @@ public Resource create(String type) throws IncorrectResourceTypeException {
         new Gold(),
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImpl.java]: hashmap :)

File: src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java [hunk 4]
```diff
@@ -41,4 +49,67 @@ public void createDefaultResources_ShouldReturnResourceList() throws JsonProcess
 
     Assert.assertEquals(expectedJson, resultJson);
   }
+
+  // region UpdateResourcesTests
+  @Test
+  public void updateResources_GoldShouldUpdateCorrectly() {
+    double testTime = 10;
+    Kingdom testKingdom = new Kingdom("testKingdom");
+    testKingdom.setResources(List.of(new Gold(), new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java]: hashmap

File: src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java [hunk 5]
```diff
@@ -41,4 +49,67 @@ public void createDefaultResources_ShouldReturnResourceList() throws JsonProcess
...
+    this.resourceService.updateResources(testKingdom);
+
+    Assert.assertEquals(expectedGold, testKingdom.getGold().getAmount());
+  }
+
+  @Test
+  public void updateResources_FoodShouldUpdateCorrectly() {
+    long testTime = 77;
+    Kingdom testKingdom = new Kingdom("testKingdom");
+    testKingdom.setResources(List.of(new Gold(), new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java]: hashmap

File: src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java [hunk 6]
```diff
@@ -41,4 +49,67 @@ public void createDefaultResources_ShouldReturnResourceList() throws JsonProcess
...
+    Assert.assertEquals(expectedFood, testKingdom.getFood().getAmount());
+  }
+
+  @Test
+  public void updateResources_FoodUpdatedAtShouldUpdateCorrectly() {
+    long updatedAt = 123456;
+    Kingdom testKingdom = new Kingdom("testKingdom");
+
+    Mockito.when(this.timeService.getCurrentTimeInMillis()).thenReturn((long) 1);
+    testKingdom.setResources(List.of(new Gold(), new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java]: hashmap

File: src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java [hunk 7]
```diff
@@ -41,4 +49,67 @@ public void createDefaultResources_ShouldReturnResourceList() throws JsonProcess
...
+
+    Assert.assertEquals(updatedAt, testKingdom.getFood().getUpdatedAt());
+  }
+
+  @Test
+  public void updateResources_GoldUpdatedAtShouldUpdateCorrectly() {
+    long updatedAt = 123456;
+    Kingdom testKingdom = new Kingdom("testKingdom");
+
+    testKingdom.setResources(List.of(new Gold(), new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/kingdomresource/ResourceServiceImplTest.java]: hashmap

File: src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java [hunk 8]
```diff
@@ -55,7 +56,9 @@ public void createDefaultResources_ShouldReturnResourceList() throws JsonProcess
   public void updateResources_GoldShouldUpdateCorrectly() {
     double testTime = 10;
     Kingdom testKingdom = new Kingdom("testKingdom");
-    testKingdom.setResources(List.of(new Gold(), new Food()));
+    testKingdom.setResources(List.of(
+        new Gold(),
+        new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java]: ) to new line

File: src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java [hunk 9]
```diff
@@ -70,46 +73,51 @@ public void updateResources_GoldShouldUpdateCorrectly() {
 
   @Test
   public void updateResources_FoodShouldUpdateCorrectly() {
-    long testTime = 77;
+    double testTime = 77;
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java]: time of wath in random unit of measure? 😄

File: src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java [hunk 10]
```diff
@@ -70,46 +73,51 @@ public void updateResources_GoldShouldUpdateCorrectly() {
...
     Kingdom testKingdom = new Kingdom("testKingdom");
-    testKingdom.setResources(List.of(new Gold(), new Food()));
+    testKingdom.setResources(List.of(
+        new Gold(),
+        new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java]: ) to new line

File: src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java [hunk 11]
```diff
@@ -70,46 +73,51 @@ public void updateResources_GoldShouldUpdateCorrectly() {
...
-  public void updateResources_FoodUpdatedAtShouldUpdateCorrectly() {
+  public void updateResources_GoldUpdatedAtShouldUpdateCorrectly() {
     long updatedAt = 123456;
     Kingdom testKingdom = new Kingdom("testKingdom");
 
-    Mockito.when(this.timeService.getCurrentTimeInMillis()).thenReturn((long) 1);
-    testKingdom.setResources(List.of(new Gold(), new Food()));
+    testKingdom.setResources(List.of(
+        new Gold(),
+        new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java]: ) to new line

File: src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java [hunk 12]
```diff
@@ -70,46 +73,51 @@ public void updateResources_GoldShouldUpdateCorrectly() {
...
   @Test
-  public void updateResources_GoldUpdatedAtShouldUpdateCorrectly() {
+  public void updateResources_FoodUpdatedAtShouldUpdateCorrectly() {
     long updatedAt = 123456;
     Kingdom testKingdom = new Kingdom("testKingdom");
 
-    testKingdom.setResources(List.of(new Gold(), new Food()));
+    testKingdom.setResources(List.of(
+        new Gold(),
+        new Food()));
```
Comment (ZaitzeV16) [inline src/test/java/com/greenfoxacademy/chamaaragonite/resource/ResourceServiceImplTest.java]: ) to new line

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/Kingdom.java [hunk 13]
```diff
@@ -71,6 +71,16 @@ public Kingdom(AppUser appUser) {
...
 
 
+  // region Getters
+  public Resource getGold() {
+    return this.getResources().get(0);
+  }
+
+  public Resource getFood() {
+    return this.getResources().get(1);
+  }
```
[the remark this answers is not in the document]
Comment (ZaitzeV16) [reply src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/Kingdom.java]: NEIN! 🇩🇪

[the remark this answers is not in the document]
Comment (mrkvcstms) [reply src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/Kingdom.java]: fight! 👊

[the remark this answers is not in the document]
Comment (csadesz) [reply src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/Kingdom.java]: Come on! 👯‍♂ 😈

**Pull request merged**